string[] names
float32[] changes
float32 fractionMaxSpeed
---
bool res